app/uk/gov/hmrc/gform/validation/ValidationUtil.scala

```diff
@@ -96,19 +98,54 @@ object ValidationUtil {
     val gFormErrors: Map[ModelComponentId, Set[String]] =
       validationResult.swap.getOrElse(Map.empty[ModelComponentId, Set[String]])
 
-    def matchComponentType(formComponent: FormComponent): FormFieldValidationResult = formComponent match {
-      case IsMultiField(_) =>
-        val valSuffixResult: List[(ModelComponentId, FormFieldValidationResult)] =
-          evaluateWithSuffix(formComponent, gFormErrors)(dataGetter)
+    def multiFieldValidationResult(
```

Comment:
extracted body of `case IsMultiField...` into function multiFieldValidationResult